pandas/core/reshape.py

```diff
@@ -1231,16 +1241,24 @@ def get_empty_Frame(data, sparse):
         return get_empty_Frame(data, sparse)
 
     codes = codes.copy()
-    if dummy_na:
+    if dummy_na and dummy_na != 'keep':
         codes[codes == -1] = len(levels)
         levels = np.append(levels, np.nan)
 
     # if dummy_na, we just fake a nan level. drop_first will drop it again
-    if drop_first and len(levels) == 1:
+    # if all nans is passed, then any column made for them will get dropped,
+    # so just check for levels <= 1
+    if drop_first and len(levels) <= 1:
         return get_empty_Frame(data, sparse)
 
     number_of_cols = len(levels)
 
+    # If all NaN values are passed, then we add a NaN column
+    # (and put NaN in it below).
+    if dummy_na == 'keep' and number_of_cols == 0:
```

Comment:
Yep! You commented on it below. :)

https://github.com/beckermr/pandas/blob/abdff7f7106778be7e1f1409759b0c5c4ba59ac5/pandas/tests/test_reshape.py#L338